### ISSUE TITLE
the proper use of CMAKE_TOOLCHAIN_FILE

### ISSUE THREAD

**Issue by donbright** [autc04/Retro68#63]
Under Raytrace the CMakeLists.txt file says that in order to build 'by itself' you need to do this:

```
# To use this example as a standalone project using CMake:
# mkdir build
# cd build
# cmake .. -DCMAKE_TOOLCHAIN_FILE=path/to/Retro68-build/toolchain/cmake/retro68.toolchain.cmake
# make
```

However I have found that this doesn't actually work, due to the add_application being not included properly in this fashion. 

I wandered around and found MacHTTP which uses Retro68, and it's CMAKE_TOOLCHAIN_FILE is slightly different, and also seems to work and solve the add_application problem. However it does produce some bizarre cmake 'use policy' warning but ... at least it builds. 

```
 cd ..
 mkdir MacHTTP-build
 cd MacHTTP-build
 cmake ../MacHTTP -DCMAKE_TOOLCHAIN_FILE=<<YOUR_PATH_TO_Retro68-build>>/toolchain/m68k-apple-macos/cmake/retro68.toolchain.cmake

```

Why does this happen? It appears that the toolchain build system actually creates two separate copies  of the retro68.toolchain.cmake file . They appear to be identical,...

```
 don@serebryanya:~/ray/bin$ find ~/r68bin/ | grep retro68.toolchain.cmake
 /home/don/r68bin/toolchain/m68k-apple-macos/cmake/retro68.toolchain.cmake
 /home/don/r68bin/build-target/cmake/retro68.toolchain.cmake
``` 

It seems that the issue is that add_application.cmake is only in one of these places:

```
 don@serebryanya:~/ray/bin$ find ~/r68bin/ | grep add_app
 /home/don/r68bin/toolchain/m68k-apple-macos/cmake/add_application.cmake
``` 

I think that may be the issue... however I am not sure of the best way to 'fix' it as it appears to be rather intricately tied to the build and the way the Samples are also built. 

Also I haven't built the PPC toolchain and am not sure of the issues involved if that were enabled. 

Thanks

**Comment by autc04** [comment]
Oh, my bad - documentation comments fixed in 2dec12faee929bafb882b557052933af49783132.

It should be
```
cmake .. -DCMAKE_TOOLCHAIN_FILE=path/to/Retro68-build/toolchain/m68k-apple-macos/cmake/retro68.toolchain.cmake
```
... the other path was outdated. There's nothing there, and apparently cmake does not report that properly as an error and instead only complains about the consequences of not having a toolchain file (it thinks you're building for your linux system and complains that `add_application` does not exist).

With this corrected path, things should "just work". I don't get policy warnings with cmake 3.11 and the Raytracer sample, but anyway, most cmake policy warnings are absolutely harmless.

> It appears that the toolchain build system actually creates two separate copies of the retro68.toolchain.cmake file . They appear to be identical,...

One is in the build directory; ignore it. The things built in the "build-target" directory (except for sample apps) get "installed" into the toolchain directory using a `make install` command, that's when the different relevant parts are put into the correct directories.
To be exact, `add_application.cmake` gets copied directly from the source directory, whereas `retro68.toolchain.cmake` is first created from `retro68.toolchain.cmake.in` (which lives in the source directory), and *then* installed (copied) to its final location.

**Comment by donbright** [comment]
Thanks
